File: Day-01/java/rainbowdashlabs/Day1.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.Arrays;
import java.util.Comparator;

public class Day1 {
    public static void main(String[] args) throws IOException {
        var calories = Arrays.stream(Files.readString(Path.of("input.txt")).split("\n\n"))
                             .map(group -> group.lines()
                                                .mapToInt(Integer::parseInt)
                                                .sum())
                             .toList();

        System.out.printf("Part 1: %d%n", calories.stream()
                                                  .mapToInt(i -> i)
                                                  .max()
                                                  .getAsInt());

        System.out.printf("Part 2: %d%n", calories.stream()
                                                  .sorted(Comparator.reverseOrder())
                                                  .limit(3)
                                                  .mapToInt(i -> i)
                                                  .sum());
    }
}

File: Day-01/kotlin/amejonah1200/day1.kt
import kotlin.io.path.Path
import kotlin.io.path.readText

fun main() {
  val calories = Path("input.txt").readText().split("\n\n").map{ it.lines().map{ it.toInt() }.sum() }
  println("Part 1: ${calories.max()}")
  println("Part 2: ${calories.sorted().takeLast(3).sum()}")
}
